tensorflow-core/tensorflow-core-api/pom.xml

```diff
@@ -0,0 +1,321 @@
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.tensorflow</groupId>
+    <artifactId>tensorflow-core</artifactId>
+    <version>2.0.0-SNAPSHOT</version>
+  </parent>
+  <artifactId>tensorflow-core-api</artifactId>
+  <packaging>jar</packaging>
+
+  <name>TensorFlow Core API Library</name>
+  <description>Platform-dependent native code and pure-Java code for the TensorFlow machine intelligence library.</description>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.bytedeco</groupId>
+      <artifactId>javacpp</artifactId>
+      <version>${javacpp.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.bytedeco</groupId>
+      <artifactId>mkl-dnn</artifactId>
+      <version>0.21-${javacpp.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+      <version>28.0-jre</version>
+      <optional>true</optional> <!-- for compilation only -->
+    </dependency>
+    <dependency>
+      <groupId>com.squareup</groupId>
+      <artifactId>javapoet</artifactId>
+      <version>1.11.1</version>
+      <optional>true</optional> <!-- for compilation only -->
+    </dependency>
```

Comment:
javapoet and guava are just required for the annotation-processor, which should be move to its own artifact (unless you find a way to make it work where you put it).

Reply:
AFAIK, it's working just fine, yes. If we uncomment the lines that start with `<!-- Bazel currently runs this`, it gets executed just fine. However, it outputs an error because it doesn't want to overwrite existing classes:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project tensorflow-core-api: Fatal error compiling: java.lang.AssertionError: javax.annotation.processing.FilerException: Attempt to recreate a file for type org.tensorflow.op.NnOps -> [Help 1]
```

Reply:
Yes, that's because you copied those files (`NnOps` & friends) in your bash script previously, if you remove that last line in the script I pointed out, it should be fine... so you are saying that the annotation-processor runs successfully directly from the core-api jar?

Reply:
Yes, that's correct. If I delete `src/gen/java`, it ends up generating them in `target/generated-sources/annotations/`. Is there a way to change that?

Reply:
Ah, here we go: https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#generatedSourcesDirectory So that should work alright...

Reply:
Ok, done! But I haven't been able to get it to output to `src/gen/java`. I tried to hack it a dozen ways, but `maven-compiler-plugin` is hard coded to not compile anything from the `generatedSourcesDirectory` directory, which makes sense I guess, but it would be nice if we could use the `includes` and `excludes` filters instead. Anyway, it's now outputting to `src/gen/annotations`. Maybe we can do the same with the others like `src/gen/javacpp` and `src/gen/ops`...